metrics.py

```diff
@@ -108,14 +112,18 @@ def calc_change_score(current_network: np.ndarray, indices: tuple):
         return -1 if current_network[indices[0], indices[1]] else 1
 
     @staticmethod
-    @njit
-    def calculate_for_sample(networks_sample: np.ndarray):
+    # @njit
+    def calculate_for_sample(networks_sample: np.ndarray | torch.Tensor):
```

Comment:
Does torch have a special type for sparse tensors? If so, we should use this as the typing hint, so there will be a chance to catch a regular tensor before we end up returning `None` because we didn't match any of the cases (the IDE will shout, or there maybe will be some informative error). Alternatively, we can validate that the tensor is sparse and raise an error if not. This is relevant everywhere the condition `if isinstance(networks_sample, torch.Tensor) and networks_sample.is_sparse:` appears.

Reply:
So there is no special type for sparse matrices. It's just a tensor with a `is_sparse` flag. Not sure about this "is_sparse" check anymore because right now every torch.Tensor is a sparse tensor. Maybe when we add regular tensors for GPU support we need a completely different type check, but right now I simply have a flag for sparse matrices